apps/passport-server/src/services/telegramService.ts

```diff
@@ -151,20 +209,51 @@ export class TelegramService {
   }
 
   private async verifyPCD(
-    serializedEdDSATicket: string
-  ): Promise<EdDSATicketPCD | null> {
-    let pcd: EdDSATicketPCD;
+    serializedZKEdDSATicket: string,
+    telegramUserId: number
+  ): Promise<ZKEdDSATicketPCD | null> {
+    let pcd: ZKEdDSATicketPCD;
 
     try {
-      pcd = await EdDSATicketPCDPackage.deserialize(
-        JSON.parse(serializedEdDSATicket).pcd
+      pcd = await ZKEdDSATicketPCDPackage.deserialize(
+        JSON.parse(serializedZKEdDSATicket).pcd
       );
     } catch (e) {
       throw new Error(`Deserialization error, ${e}`);
     }
 
-    // Right now, we are only verifying that the PCD is authentic
-    if (await EdDSATicketPCDPackage.verify(pcd)) {
+    // this is very bad but i am very tired
+    // hardcoded eventIDs and signing keys for SRW
+    let signerMatch = false;
+    let eventIdMatch = false;
+    if (process.env.PASSPORT_SERVER_URL === "http://localhost:3002") {
+      eventIdMatch = true;
+      signerMatch = true;
+    } else if (process.env.PASSPORT_SERVER_URL?.includes("staging")) {
+      eventIdMatch =
+        pcd.claim.partialTicket.eventId ===
+        "3fa6164c-4785-11ee-8178-763dbf30819c";
+      signerMatch =
+        pcd.claim.signer[0] ===
+          "7cf4d97878d663502339c2baae74b12dcdd229279a9f6bfc83b167e808a32d26" &&
+        pcd.claim.signer[1] ===
+          "c5a04b56d0f2d6b1ec10aa1b17298e31b4a087bdabd4a75d9523779e7dca5a17";
+    } else {
+      eventIdMatch =
+        pcd.claim.partialTicket.eventId ===
+        "264b2536-479c-11ee-8153-de1f187f7393";
+      signerMatch =
+        pcd.claim.signer[0] ===
+          "a7da882cd090c14a62b70cf07010c1cabb373b17ebd2d120c9de039ceaedfa24" &&
+        pcd.claim.signer[1] ===
+          "509e44aa56e97a34e9a54534ef79d484d801757720d18ed872e93dd9de126b09";
+    }
+    if (
+      (await ZKEdDSATicketPCDPackage.verify(pcd)) &&
+      pcd.claim.watermark === telegramUserId.toString() &&
```

Comment:
if we're gonna be checking watermark == TG user ID anyway here, we can probably also remove the `/:id` parameter from the `/verify/` endpoint. The TG user ID is self-evident from the ZKEdDSA PCD itself.

Reply:
punting this for after SRW also makes sense since it's really a nit